File: CS/E1984/MyTreeListSerializeHelper.cs
using DevExpress.Utils.Serializing;
using DevExpress.XtraTreeList;
using System.IO;

namespace E1984 {
    public static class MyTreeListSerializeHelper
    {
        public static string DefaultFileName = "test.xml";
        public static TreeList SelectedTreeList;

        public static void SaveTreeList()
        {
            SaveTreeList(SelectedTreeList, DefaultFileName);
        }

        public static void RestoreTreeList()
        {
            RestoreTreeList(SelectedTreeList, DefaultFileName);
        }



        static void SaveTreeList(TreeList treeList, string fileName)
        {
            treeList.SaveLayoutToXml(fileName);
            SaveConditions(treeList, "Conditions" + fileName);
        }

        private static void SaveConditions(TreeList treeList, string fileName)
        {
            XmlXtraSerializer serializer = new XmlXtraSerializer();
            using (FileStream stream = new FileStream(fileName, FileMode.OpenOrCreate))
            {
                try
                {
                    TreeListFormatConditions collection = new TreeListFormatConditions(treeList.FormatConditions);
                    serializer.SerializeObject(collection, stream, "FormatConditions");
                }
                finally { stream.Close(); }
            }
        }

        static void RestoreTreeList(TreeList treeList, string fileName)
        {
            treeList.RestoreLayoutFromXml(fileName);
            RestoreConditions(treeList, "Conditions" + fileName);
        }


        private static void RestoreConditions(TreeList treeList, string fileName)
        {
            if (!File.Exists(fileName)) return;
            XmlXtraSerializer serializer = new XmlXtraSerializer();
            using (FileStream stream = new FileStream(fileName, FileMode.Open))
            {
                try
                {
                    TreeListFormatConditions collection = new TreeListFormatConditions();
                    serializer.DeserializeObject(collection, stream, "FormatConditions");
                    collection.Conditions.CopyToStyleFormats(treeList.FormatConditions);
                    treeList.Refresh();
                }
                finally { stream.Close(); }
            }
        }
    }
}

File: VB/E1984/MyTreeListSerializeHelper.vb
Imports DevExpress.Utils.Serializing
Imports DevExpress.XtraTreeList
Imports System.IO

Namespace E1984
	Public Module MyTreeListSerializeHelper
		Public DefaultFileName As String = "test.xml"
		Public SelectedTreeList As TreeList

		Public Sub SaveTreeList()
			SaveTreeList(SelectedTreeList, DefaultFileName)
		End Sub

		Public Sub RestoreTreeList()
			RestoreTreeList(SelectedTreeList, DefaultFileName)
		End Sub



		Private Sub SaveTreeList(ByVal treeList As TreeList, ByVal fileName As String)
			treeList.SaveLayoutToXml(fileName)
			SaveConditions(treeList, "Conditions" & fileName)
		End Sub

		Private Sub SaveConditions(ByVal treeList As TreeList, ByVal fileName As String)
			Dim serializer As New XmlXtraSerializer()
			Using stream As New FileStream(fileName, FileMode.OpenOrCreate)
				Try
					Dim collection As New TreeListFormatConditions(treeList.FormatConditions)
					serializer.SerializeObject(collection, stream, "FormatConditions")
				Finally
					stream.Close()
				End Try
			End Using
		End Sub

		Private Sub RestoreTreeList(ByVal treeList As TreeList, ByVal fileName As String)
			treeList.RestoreLayoutFromXml(fileName)
			RestoreConditions(treeList, "Conditions" & fileName)
		End Sub


		Private Sub RestoreConditions(ByVal treeList As TreeList, ByVal fileName As String)
			If Not File.Exists(fileName) Then
				Return
			End If
			Dim serializer As New XmlXtraSerializer()
			Using stream As New FileStream(fileName, FileMode.Open)
				Try
					Dim collection As New TreeListFormatConditions()
					serializer.DeserializeObject(collection, stream, "FormatConditions")
					collection.Conditions.CopyToStyleFormats(treeList.FormatConditions)
					treeList.Refresh()
				Finally
					stream.Close()
				End Try
			End Using
		End Sub
	End Module
End Namespace
